src/python/pants/backend/native/tasks/native_compile.py

```diff
@@ -185,10 +187,20 @@ def get_compiler(self):
   def _compiler(self):
     return self.get_compiler()
 
+  def get_third_party_include_dirs(self):
+    inc_dir = []
+    tp_files_product = self.context.products.get_data(NativeExternalLibraryFetch.NativeExternalLibraryFiles)
```

Comment:
Again, it's usually much more clear to fetch products in the body of the `execute()` method, and if that is done this method can be removed entirely as well (making it into a separate method actually obscures the logic here a little).

Reply:
✅